### PR TITLE
Reuse react-native-layout-provider

### PR DESCRIPTION
**Pull request by jhen0409**
Related to https://github.com/machadogj/react-native-layout-tester/issues/6#issuecomment-231200031. Split context & getLayout to [react-native-layout-provider](https://github.com/jhen0409/react-native-layout-provider) and reuse it.

### CODE REVIEW

**Comment by machadogj** [comment]
@jhen0409 this is really good! I like how you unit tested the layout-provider 👍

**Comment by machadogj** [comment]
We do have to update the readme section to import from react-native-layout-provider now. I'll add an issue to not forget about it.

File: index.js [hunk 1]
```diff
@@ -144,10 +100,8 @@ export default class LayoutTester extends Component {
                 width: config.width
             },
             portrait: config.portrait || this.state.portrait
-        };
-        this.setState(newState, () => {
-            this.listeners.forEach(listener => listener(this.getState()));
         });
```
Comment (jhen0409) [inline index.js:103]: This should be `};`, sorry @machadogj.

File: index.js [hunk 2]
```diff
@@ -163,11 +117,9 @@ export default class LayoutTester extends Component {
         };
         this.setState(newState, () => {
             if (this.props.viewportChanged) {
-                this.props.viewportChanged(this.getState());
+                this.props.viewportChanged(newState));
```
Comment (jhen0409) [inline index.js:120]: And this.

Comment (machadogj) [inline index.js:120]: Done, thx.

File: index.js (hunk 1 above)
Comment (machadogj) [inline index.js:103]: Done, thx!

**Pull request merged**